app/services/webhooks/base_service.rb

```diff
@@ -75,13 +75,20 @@ def send_webhook(webhook, webhook_endpoint, payload)
         .perform_later(webhook_type, object, options, webhook.id)
     end
 
-    def generate_headers(payload)
+    def generate_headers(webhook_endpoint, payload)
+      signature = case webhook_endpoint.signature
+                  when :jwt
+                    jwt_signature(payload)
+                  when :hmac
+                    hmac_signature(payload)
+      end
+
       [
-        'X-Lago-Signature' => generate_signature(payload),
+        'X-Lago-Signature' => signature,
```

Comment:
It will be a breaking change.
Should we add a flag exposing which algo was used to generate the signature?
It could help for a smooth transition

Reply:
It won't be a breaking change right? The migration will set existing records to signature `0`, which is JWT. 

But I agree that adding a `X-Lago-Signature-Algorithm` would be nice.